src/Redistribution.sol

```diff
@@ -360,39 +407,46 @@ contract Redistribution is AccessControl, Pausable {
         ChunkInclusionProof calldata entryProof2,
         ChunkInclusionProof calldata entryProofLast
     ) external whenNotPaused {
-        winner = winnerSelection();
+        winnerSelection();
 
+        if (winner.owner != msg.sender) {
+            revert SenderNotWinner();
+        }
+
+        Reveal memory winnerSelected = winner;
```

Comment:
why not before the if statement?
```suggestion
       Reveal memory winnerSelected = winner;

        if (winnerSelected.owner != msg.sender) {
            revert SenderNotWinner();
        }
```

Reply:
It could be, generally, we want to **revert** to fire ASAP as we save a bit of gas for trx that way, so just a good way to put it there before all others. Its small difference in the end but I don't see reason to put it before?

Reply:
because it will read out from storage which is more expensive, isn't it? the memory read is 3-5 gas, the storage read is 200-800